actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java

```diff
@@ -0,0 +1,169 @@
+import org.sunbird.cassandra.CassandraOperation;
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.response.Response;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.models.util.ProjectUtil;
+import org.sunbird.common.models.util.StringFormatter;
+import org.sunbird.common.models.util.datasecurity.EncryptionService;
+import org.sunbird.common.request.Request;
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.helper.ServiceFactory;
+import org.sunbird.learner.util.Util;
+
+public class AuthenticationHelper {
+
+  private static CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+  private static EncryptionService encryptionService =
+      org.sunbird.common.models.util.datasecurity.impl.ServiceFactory.getEncryptionServiceInstance(
+          null);
+
+  public static void validateWithUserId(Request reqObj) {
+    if (null != reqObj.getContext().get(JsonKey.IS_AUTH_REQ)
+        && Boolean.parseBoolean((String) reqObj.getContext().get(JsonKey.IS_AUTH_REQ))) {
+      String ctxtUserId = (String) reqObj.getContext().get(JsonKey.USER_ID);
+      String userId = getUserIdFromExtIdAndProvider(reqObj);
+      if ((!StringUtils.isBlank(userId)) && (!userId.equals(ctxtUserId))) {
+        throw new ProjectCommonException(
+            ResponseCode.unAuthorized.getErrorCode(),
+            ResponseCode.unAuthorized.getErrorMessage(),
+            ResponseCode.UNAUTHORIZED.getResponseCode());
+      }
+    }
+  }
+
+  private static String getUserIdFromExtIdAndProvider(Request reqObj) {
+    String userId = "";
+    if (null != reqObj.getRequest().get(JsonKey.USER_ID)) {
+      userId = (String) reqObj.getRequest().get(JsonKey.USER_ID);
+    } else {
+      userId = (String) reqObj.getRequest().get(JsonKey.ID);
+    }
+    if (StringUtils.isBlank(userId)) {
+      String extId = (String) reqObj.getRequest().get(JsonKey.EXTERNAL_ID);
+      String provider = (String) reqObj.getRequest().get(JsonKey.EXTERNAL_ID_PROVIDER);
+      String idType = (String) reqObj.getRequest().get(JsonKey.EXTERNAL_ID_TYPE);
+      Map<String, Object> user = getUserFromExternalId(extId, provider, idType);
+
+      if (user != null && !user.isEmpty()) {
+        userId = (String) user.get(JsonKey.ID);
+      } else {
+        throw new ProjectCommonException(
+            ResponseCode.invalidParameter.getErrorCode(),
+            ProjectUtil.formatMessage(
+                ResponseCode.invalidParameter.getErrorMessage(),
+                StringFormatter.joinByAnd(
+                    StringFormatter.joinByComma(JsonKey.EXTERNAL_ID, JsonKey.EXTERNAL_ID_TYPE),
+                    JsonKey.EXTERNAL_ID_PROVIDER)),
+            ResponseCode.CLIENT_ERROR.getResponseCode());
+      }
+    }
+    return userId;
+  }
+
+  @SuppressWarnings({"unchecked"})
+  public static Map<String, Object> getUserFromExternalId(
```

Comment:
Define this in UserExternalIdentityDao.getUserByExternalId.

Reply:
This was also not used - so got removed along with class.